Tests/NIOCoreTests/ByteBufferTest.swift

```diff
@@ -1878,6 +1889,29 @@ class ByteBufferTest: XCTestCase {
         XCTAssertEqual(expected, actual)
     }
 
+    func testWriteHexEncodedBytes() throws {
+        var buffer = ByteBuffer(hexEncodedBytes: "68 65 6c 6c 6f 20 77 6f 72 6c 64 0a")
```

Comment:
This doesn't `throw` and is non-`Optional`, what happens if the input isn't a valid hex string? Examples

- `686`
- `686hello`
- `!.=/`
- `1 2 3 4 5 6 7 8` (this one could be argued is valid as `12 34 56 78`

@Lukasa WDYT is the right shape of the API here?

Reply:
Once my above comment about how to implement it is answered. I wil add throws for these edge cases.

Reply:
I suspect we should throw.

Reply:
Can you let me know if it looks OK. I have made everything throw including the initialiser and the write method.